### ISSUE TITLE
Move account to a different instance, site verification fails

### ISSUE THREAD

**Issue by p37307** [mastodon/mastodon#21336]
### Steps to reproduce the problem

1. Have a Mastodon account with a verified "rel=me" website on an instance.
2. Move instances and setup profile correctly with the website in the same table row box as in Profile metadata, edit website with new "rel=me" code 
3. Site doesn't verify
4. Move website to a different 'profile metadata' table row and the site verifies.
...

### Expected behaviour

Site should verifiy

### Actual behaviour

Doesn't verify unless you change placements

### Detailed description

Website doesn't verify after you move instances unless you enter it into a different placement in the 'Profile metadata' setup on profile page.

I'm not sure if this has something to do with the behind-the-scenes migration process or not.

### Specifications

Mastodon 4.0.2

**Comment by afontenot** [comment]
> Move instances and setup profile correctly with the website link in the same table row box as in Profile metadata, edit website with new "rel=me" code

If you did the steps in this order, that's probably the problem. Mastodon has no way of knowing that you edited your site, so it doesn't check the verification (which previously failed because you hadn't changed the link yet). To get it to check the verification again, you have to make some change (e.g. moving the website to a different row, as you did), and then resave it. You could also change "Website" to "website" and then back again, or anything really.

**Comment by p37307** [comment]
> Mastodon has no way of knowing that you edited your site,

It was a new profile on a new instance, and I added the website and link to it and used the new instance re=me link code. I assume the new instance would have known, as "Mastodon," to check for the verification unless there is a central database somewhere independent of the thousands of individual servers keeping dibs on verified websites. I don't know enough about the inner and outer workings to presume that. I doubt that is the case.

It could have been a failed worker process or something like that. I should edit my post to say that more than 24-hours went by but it verified almost immediately after I moved slots.

Still worthy of someone else testing and see if this is an issue happening to others.

**Comment by ineffyble** [comment]
Mastodon migration shouldn't have any relevance here because you manually set up the new profile.

**Comment by p37307** [comment]
@afontenot, let me see if I understand what you are saying.

The instruction on Mastodon says, 
` Verification
You can verify yourself as the owner of the links in your profile metadata. For that, the linked website must contain a link back to your Mastodon profile. The link back must have a rel="me" attribute. The text content of the link does not matter. Here is an example: <a rel="me" href="https://<INSTANCE URL>/@<USERNAME">Mastodon</a>
`

But you are saying it should be, place the  rel=me link on your website first, _then_ add it to the profile and save the profile.

If this is the case, that might explain why so many others can't get their sites to verify.

**Comment by afontenot** [comment]
> But you are saying it should be, place the rel=me link on your website first, _then_ add it to the profile and save the profile.

That's correct. This part kinda suggests it maybe, but I agree that it's too ambiguous:

> If you put an HTTPS link in your profile metadata, Mastodon checks if that link resolves to a web page that links back to your Mastodon profile with a special `rel=me` attribute.

Actually, maybe the real issue is that nothing specifies *when* exactly the validation occurs. Does it occur once a day automatically? Every time you save your profile? Whenever you edit the metadata item?

The actual answer is surprising: federated instances on which at least one user follows you will do the verification *every time* the profile is saved, even without changes. Your own instance only checks the verification when you edit the metadata item. Federated instances where no one follows you do the verification on the fly, whenever someone views your profile the first time.
